code/datums/chatmessage.dm

```diff
@@ -555,6 +579,45 @@
 	var/duration = BALLOON_TEXT_TOTAL_LIFETIME(duration_mult)
 	fadertimer = addtimer(CALLBACK(src, PROC_REF(end_of_life)), duration, TIMER_STOPPABLE|TIMER_DELETE_ME, SSrunechat)
 
+/**
+ * Approximates the chatmesseges width based on cached widths of each char.
+ * If the character is not found in this cache we assume the worst and add the highest possible value.
+ *
+ * Arguments:
+ * * string - string to measure width
+ * * font size - font size that the displayed string will be in, used to calculate font size multiplier
+ * * is_bold - passed if the font is bold, the approximation takes into account additional width of the font
+ * * has_icon - text has an icon, which adds extra 8 pixels
+ */
+/datum/chatmessage/proc/approx_str_width(string, font_size = DEFAULT_FONT_SIZE, is_bold = FALSE, has_icon = FALSE)
+	var/value = 0
+	var/index = NORMAL_FONT_INDEX
+	if(font_size == WHISPER_FONT_SIZE)
+		index = SMALL_FONT_INDEX
+	else if(font_size == BIG_FONT_SIZE)
+		index = BIG_FONT_INDEX
+
+	var/i = 1
+	while(i <= length(string))
+		var/list/letters = SSrunechat.letters[string[i]]
```

Comment:
For how many times this gets called, using an assoc lookup is going to be slow. If you have a sequence of characters yoy can just use an int lookup with base offset

Reply:
If you are sure converting char to int will be faster than accessing this list I can change it

Reply:
Changed